### ISSUE TITLE
Global Leaderboard

### ISSUE THREAD

**Issue by Shivansh-007** [python-discord/sir-lancebot#627]
## Description
Add a global points leader-board, for the points/coins which are earned by users while playing a game like Trivia Quiz, Tic Tac Toe, Connect 4, etc. Currently each game has its own leader-board, and not a combined one together for all games. In short, combining all the leader-boards into one.

## Proposed Implementation

- [ ] Make a new redis cache in `bot/bot.py` called `Leaderboard` (say).
- [ ] Make a points system in all the games which make use of a leaderboard namely
      • Battleship
      • Connect 4
      • Riddle
      • Eggquiz
      • Minesweeper
      • Tic Tac Toe
      • Quiz
      • Snake Quiz
      • Snakes And Ladders

- [ ] Make a cog for for viewing the leaderboard in a pretty way, we can make the coins maybe `:lemon_coin:` or `:gurkan_coin:` or `:gem_red:` (can be discussed more in `#media-branding`)
- [ ] Make `add_point` and `remove_points` utils functions, would be helpful when users earn points.

## Would you like to implement this yourself?
- [ ] I'd like to implement this feature myself
- [x] Anyone can implement this feature

**Comment by dawnofmidnight** [comment]
The points scoring for the rest of the games would definitely have to be normalized. That might be as simple as changing 5 points to 50 in one game to match the rest. Either way, this would be a lot of work, but it's a really good idea!

**Comment by Akarys42** [comment]
I like this idea. Maybe we should introduce a maximum amount of points earned per game and per day, so spamming Quizzes alone won’t get you to the top of the board?

**Comment by Shivansh-007** [comment]
Yeah, that sounds a good idea, maximum per day points per game. Users would be able to play the game but won’t earn points for them.

**Comment by jchristgit** [comment]
@ammaralzeno sure thing, assigned you
